src/Microsoft.VisualStudio.Services.Agent/TraceManager.cs

```diff
@@ -71,6 +71,10 @@ private void Dispose(bool disposing)
         private Tracing CreateTraceSource(string name)
         {
             SourceSwitch sourceSwitch = Switch;
+
+            // Flush it to avoid loosing the log if the process is killed
+            Trace.AutoFlush = true;
```

Comment:
The logs never get flushed if the app is terminated. (Right now if it runs as service its always killed, as its not litening to Ctrl+C event) You  have to press Ctrl+c on the terminal while running as interactive process only then logs gets flushed..